prog/encoding_test.go

```diff
@@ -30,13 +30,17 @@ func TestSerializeData(t *testing.T) {
 			for i := range data {
 				data[i] = byte(r.Intn(256))
 			}
+			if 0 < len(data) && data[0] == '$' {
```

Comment:
I don't think this is necessary. The `data` is deserialized data, e.g. a file name in a program, it can contain any symbols. `$` should be fine, since it's serialized first.